### PR TITLE
extend slack recipe

### PR DESCRIPTION
**Pull request by drieshooghe**
add possibility to pass username, pretext and icon_emoji as attachments

| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | N/A

### CODE REVIEW

File: recipe/slack.php [hunk 1]
```diff
@@ -31,10 +31,13 @@
     }
 
     $attachment = [
+        'username' => get('slack_username'),
         'title' => get('slack_title'),
+        'pretext' => get('slack_pretext'),
```
Comment (antonmedv) [inline recipe/slack.php]: What if user didn't specified slack_pretext? Same for slack_icon.

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: recipe/slack.php (hunk 1 above)
Comment (drieshooghe) [reply recipe/slack.php]: My apologies, I overlooked this. According to the slack api documentation the value of each property can be an empty string (" "). But I gave the icon and username a default value.

**Comment by antonmedv** [comment]
Please show example of how  now notify looks like

**Comment by drieshooghe** [comment]
screen shot 2018-01-26 at 13 03 14

**Comment by antonmedv** [comment]
Okay, but logo should be Deployer's Logo by default.

**Comment by drieshooghe** [comment]
To get a result like the image I sent, the username and icon need to be outside of the attachment, that's why I adjusted the array. Missing values get an empty string and are thus ignored by slack, so the change isn't breaking.

**Comment by lucasmezencio** [comment]
@drieshooghe I really like this PR!
I was about to create a PR when I figured out this. But I think it could be more 'extensible'.
I think that the user could be able to override all options such as attachments, messages, icons, and so  on.

Few other things I missed in this PR are:
- PRS-1/2 compliance;
- Updated documentation;

I'm not intending for being rude or something like thst pointing these things, I would just really like to help you with this PR.
If @antonmedv accept it as is, I can send another PR with these changes. :)

**Pull request closed without merging**